frontend/src/components/MapView.js

```diff
@@ -12,40 +16,196 @@ const mapStyles = {
+      // The event object
+      const event = dataSnapshot.val();
+      // If the state has the event then update the change
+      if (this.state.allEvents[eventId] !== undefined) {
+	this.updateEvent(eventId, event);
+      } else {
+	// If the state doesnt have the event, add the event to the map
+	this.setState(prevState => ({
+  	  allEvents: [...prevState.allEvents, event]
+	}));
+	this.forceUpdate();
+      }
+    });
+  }
+
+  // Updates the event info box and updates the map in memory
+  updateEvent(eventId, event) {
+    console.log("updatesEvent")
+    // TODO: Update event info box
+    this.state.allEvents.push({
+      key: eventId,
+      value: event
+    }); 
+    this.setState({
+      allEvents: this.state.allEvents,
+    });
+  }
+
+  loadArticle(article) {
+    console.log(article);
+    console.log("loadArticle:");
+    if (article.locationObject) {
+      this.plusCodeGlobalCode = article.locationObject.plus_code.global_code;
+      this.setState({allEvents: []});
+    }
+  }
+
+  addInfoBoxEvent(event) {
+    return this.getInfoBox(event);
+  }
+
+  getInfoBox(event) {
+    console.log("infobox");
+    var location = this.getCoords(event.location);
+    var lat = parseFloat(location[0]);
+    var lng = parseFloat(location[1]);
+
+    var length = 0;
+    var imageUrl = "";
+    if (event.imagePaths != null) {
+      length = event.imagePaths.length;
+    }
+
+    if (length > 0) {
+      imageUrl = event.imagePaths.slice(1, length - 2);
+      imageUrl = imageUrl.split(",")[0];
+    }
+
+    return(
+      <InfoWindow
+	    visible={this.state.showInfoWindows}
+	    position={{lat: lat, lng: lng}}>
+	  <Card border="light">
+      	<Card.Img variant="right" src={imageUrl} />
+      	  <Card.Body>
+      	    <Card.Title>{event.eventName}</Card.Title>
+      	    <Card.Text>{event.description}</Card.Text>
+      	  </Card.Body>
+	  </Card>
+      </InfoWindow>
+    );
+  }
 
-    this.state = {};
+  // Gets coordinate from string of location. Element 0 is latitude and 1 is longitude
+  getCoords(location) {
+    var coords = "";
+    var length = 0;
+    if (location != null) {
+     length = location.length;
+    }
+    if (length > 0) {
+      coords = location.slice(1, length-2);
+      coords = coords.split(",");
+    }
+    return coords;
   }
 
-  render() {
+  onReady = () => {
+    this.setState({
+      showInfoWindows: true
+    });
+  }
+
+render() {
     return (
-      <div>
+      <div  className="mapView">
         {this.props.articles.map(article => {
           return (
             <Map
+	      id="map"
               key={article.toString()}
               google={this.props.google}
               zoom={17}
+	          onReady={this.onReady}
               style={mapStyles}
               initialCenter={{
-                lat: article.location.lat(),
-                lng: article.location.lng()
+                lat: 39.9522188,
```

Comment:
I think we can go back to having lat and lng be article.location.lat() and article.location.lng().